source/gameengine/GameLogic/Joystick/SCA_JoystickEvents.cpp

```diff
@@ -37,45 +37,29 @@
 #endif
 
 #ifdef WITH_SDL
-void SCA_Joystick::OnAxisMotion(SDL_Event* sdl_event)
+void SCA_Joystick::OnAxisEvent(SDL_Event* sdl_event)
 {
-	if (sdl_event->jaxis.axis >= JOYAXIS_MAX)
+	if (!this || sdl_event->caxis.axis >= JOYAXIS_MAX)
```

Comment:
Can you explain the `!this` condition ? I guess is to test if the joystick is non NULL but it should be done before call the `OnAxisEvent` function.